### PR TITLE
Bannerflow amp ad implementation

### PR DESCRIPTION
**Pull request by joakim-nordicfactory**

### CODE REVIEW

**Comment by joakim-nordicfactory** [comment]
I signed it!

File: ads/banneflow.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/banneflow.md]: nit: 2017

File: ads/banneflow.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+limitations under the License.
+-->
+
+# Bannerflow
+
+## Example
+
+```html
+<amp-ad type="bannerflow"
+        width="400"
```
Comment (zhouyx) [inline ads/banneflow.md]: nit: indentation

File: ads/banneflow.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        height="250"
+        src="https://embed.bannerflow.com/5965e2cae9b2bcac18d784aa"></amp-ad>
```
Comment (zhouyx) [inline ads/banneflow.md]: nit: </amp-ad> in a new line please

File: ads/bannerflow.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/bannerflow.js]: nit: 2017

File: ads/bannerflow.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import {
+  writeScript,
+  validateSrcPrefix,
+} from '../3p/3p';
+
+
+export function bannerflow(global, data) {
+  const src = data.src;
```
Comment (zhouyx) [inline ads/bannerflow.js]: please validate data.src before calling `#validateSrcPrefix()`. You can use the `#validateData()` function

File: ads/banneflow.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+## Configuration
+For semantics of configuration, please see ad network documentation.
```
Comment (zhouyx) [inline ads/banneflow.md]: Could you link to the document here?

File: ads/_config.js [hunk 7]
```diff
@@ -207,6 +207,17 @@ export const adConfig = {
...
+  bannerflow: {
+    prefetch: [
+      'https://scripts.host.bannerflow.com/1.0.4/bf.min.js',
+      'https://scripts.host.bannerflow.com/1.1.4/render.min.js',
+    ],
+    preconnect: [
+      'https://resources.host.bannerflow.com',
+      'https://banners.host.bannerflow',
+    ],
+  },
```
Comment (zhouyx) [inline ads/_config.js]: We have a renderStart [API](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis) that help to improve user experiences. We highly recommend implementing it. Please let me know if there's any question.

File: ads/banneflow.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+<!---
```
Comment (zhouyx) [inline ads/banneflow.md]: Please also add bannerflow to `amp-ad.md` and link to this document.

File: ads/_config.js (hunk 7 above)
Comment (joakim-nordicfactory) [reply ads/_config.js]: We will implement this in our ad scripts in the near future.

**Comment by joakim-nordicfactory** [comment]
How do I solve this issue: "CLAs are signed, but unable to verify author consent"?

**Comment by joakim-nordicfactory** [comment]
I used a different email address in github, and I have now changed it so the CLA should be accepted! Please let the bot check again...

**Comment by joakim-nordicfactory** [comment]
I now have one individual and one corporate agreement...

**Comment by joakim-nordicfactory** [comment]
Please let the bot check the new CLA thanks

**Comment by zhouyx** [comment]
@joakim-nordicfactory Please rebase to latest master. 
Your pull request contains commit from other authors. (124 files changed) Please rebase to only include your commit and CLA check should resolve.

File: ads/bannerflow.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+limitations under the License.
+-->
+
+# Bannerflow
+
+## Example
+
+```html
+<amp-ad type="bannerflow"
+        width="400"
```
Comment (zhouyx) [inline ads/bannerflow.md]: nit: indentation 4 please.

**Comment by zhouyx** [comment]
@joakim-nordicfactory Please rebase to latest master to fix the travis build. So that I can merge the Pull Request. Thanks!

**Comment by zhouyx** [comment]
@joakim-nordicfactory Please rebase : ) Trying to merge this Pull Request

**Pull request closed without merging**